cmd/performance-profile-creator/cmd/root.go

```diff
@@ -55,45 +55,90 @@ type ProfileData struct {
 // ClusterData collects the cluster wide information, each mcp points to a list of ghw node handlers
 type ClusterData map[*machineconfigv1.MachineConfigPool][]*profilecreator.GHWHandler
 
-// rootCmd represents the base command when called without any subcommands
-var rootCmd = &cobra.Command{
-	Use:   "performance-profile-creator",
-	Short: "A tool that automates creation of Performance Profiles",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
-		cluster, err := getClusterData(mustGatherDirPath)
-		if err != nil {
-			return fmt.Errorf("failed to parse the cluster data: %v", err)
-		}
+func init() {
+	log.SetOutput(os.Stderr)
+	log.SetFormatter(&log.TextFormatter{
+		DisableTimestamp: true,
+	})
+}
 
-		info, err := strconv.ParseBool(cmd.Flag("info").Value.String())
-		if err != nil {
-			return fmt.Errorf("failed to parse the info flag: %v", err)
-		}
-		if info {
-			showClusterData(cluster)
+// NewRootCommand returns entrypoint command to interact with all other commands
+func NewRootCommand() *cobra.Command {
+	pcArgs := &ProfileCreatorArgs{
+		UserLevelNetworking: pointer.BoolPtr(false),
+	}
+
+	var requiredFlags []string
+
+	root := &cobra.Command{
+		Use:   "performance-profile-creator",
+		Short: "A tool that automates creation of Performance Profiles",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !cmd.Flag("must-gather-dir-path").Changed {
+				return fmt.Errorf("missing cluster data")
+			}
+
+			mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
+			cluster, err := getClusterData(mustGatherDirPath)
+			if err != nil {
+				return fmt.Errorf("failed to parse the cluster data: %v", err)
+			}
+
+			info, err := strconv.ParseBool(cmd.Flag("info").Value.String())
+			if err != nil {
+				return fmt.Errorf("failed to parse the info flag: %v", err)
+			}
+			if info {
+				showClusterData(cluster)
+				return nil
+			}
+
+			if !haveRequiredFlags(cmd, requiredFlags...) {
+				return fmt.Errorf("missing required flags: %s", strings.Join(requiredFlags, ", "))
```

Comment:
This will return all the `requiredFlags` even if the user specifies a subset of them. We should either improve the error messge and say "please specify all the required flags:" or indicate the missing flag that the user did not specify.

Reply:
good idea, done.